src/Bearded.TD.Generators/Events/EventHandlerGenerator.cs

```diff
@@ -0,0 +1,187 @@
+                .WhereNotNull()
+                .Collect()
+                .SelectMany(static (methods, _) => groupByClass(methods))
+                .WithComparer(new OrderIndependentClassWithHandlersComparer());
+
+        context.RegisterSourceOutput(handlerMethods, generateSource);
+    }
+
+    private static bool isValidHandlerMethod(SyntaxNode node)
+    {
+        // Check the method
+        if (node is not MethodDeclarationSyntax method ||
+            method.ParameterList.Parameters.Count != 1)
+        {
+            return false;
+        }
+
+        // Check the parent class
+        if (method.Parent is not ClassDeclarationSyntax { Modifiers: var mods } ||
+            !mods.Any(static m => m.IsKind(SyntaxKind.PartialKeyword)))
+        {
+            return false;
+        }
+
+        return true;
+    }
+
+    private static HandlerMethod? handlerMethodFromContext(GeneratorAttributeSyntaxContext context)
+    {
+        var method = (IMethodSymbol) context.TargetSymbol;
+        var semanticModel = context.SemanticModel;
+
+        var methodName = method.Name;
+        var eventParameter = method.Parameters[0];
+        if (eventParameter.Type.AllInterfaces
+            .Select(TypeName.From)
+            .All(name => name.FullName.Name != eventBaseTypeName))
+        {
+            return null;
+        }
+
+        var eventType = TypeName.From(eventParameter.Type);
+
+        if (semanticModel.GetDeclaredSymbol(context.TargetNode.Parent!) is not ITypeSymbol parentClass)
+        {
+            return null;
+        }
+
+        var componentType = TypeName.From(parentClass);
+        var componentNamespace = Namespace.From(parentClass.ContainingNamespace);
+
+        return new HandlerMethod(componentType, componentNamespace, methodName, eventType);
+    }
+
+    private static IEnumerable<ClassWithHandlers> groupByClass(ImmutableArray<HandlerMethod> methods)
+    {
+        return methods.GroupBy(m => m.ComponentType)
+            .Select(group => new ClassWithHandlers(
+                group.Key,
+                group.First().Namespace,
+                group.OrderBy(m => m.EventType.ShortName.Name).ToImmutableArray()))
+            .OrderBy(c => c.Name.ShortName.Name);
+    }
+
+    private static void generateSource(
+        SourceProductionContext context, ClassWithHandlers classWithHandlers)
+    {
+        var methods = classWithHandlers.Handlers;
+
+        var aliases = new TypeAliases();
+        aliases.AddRange(methods.Select(m => m.EventType));
+
+        // TODO: add newlines
+        var listenerList =
+            string.Join(", ", methods.Select(m => $"IListener<{aliases[m.EventType]}>"));
+
+        var source =
+$$"""
+// This file is generated by Bearded.TD.Generators
+{{Foreach(0, defaultNamespaces, n => $"using {n};")}}
+{{Foreach(0, aliases.Where(a => a.FullName.Name != a.Alias.Name).OrderBy(a => a.Alias.Name), a => a.ToString())}}
```

Comment:
If this will always be the same, perhaps we can add this formatting code as a method on the aliases type?

Reply:
On second thought, probably better to add it to the SourceTemplates.